Competition2017/src/competition/subsystems/drive/commands/FieldOrientedTankDriveWithJoystick.java

```diff
@@ -0,0 +1,51 @@
+package competition.subsystems.drive.commands;
+
+import java.util.function.DoubleSupplier;
+
+import com.google.inject.Inject;
+
+import competition.operator_interface.OperatorInterface;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+import xbot.common.controls.sensors.XJoystick;
+import xbot.common.math.PIDFactory;
+import xbot.common.math.PIDManager;
+
+public class FieldOrientedTankDriveWithJoystick extends BaseDriveCommand {
+
+    private final PoseSubsystem poseSubsystem;
+    
+    private final XJoystick leftJoystick;
+    
+    private final PIDManager headingDrivePid;
+    public final double defaultPValue = 1 / 80d;
+
+    protected DoubleSupplier targetHeadingSupplier;
+    
+    @Inject
+    public FieldOrientedTankDriveWithJoystick(
+            OperatorInterface oi, 
+            DriveSubsystem driveSubsystem, 
+            PoseSubsystem poseSubsystem,
+            PIDFactory pidFactory) {
+        super(driveSubsystem);
+        this.poseSubsystem = poseSubsystem;
+        this.leftJoystick = oi.leftJoystick;
+        headingDrivePid = pidFactory.createPIDManager("Rotate to heading", defaultPValue, 0, 0, 0, 1, -1, 3, 3.0 / 20, 1);
```

Comment:
This shouldn't be called "rotate to heading... it's a different set of properties.